### PR TITLE
List users groups

### PR DESCRIPTION
**Pull request by NDBman**
Wasn't able to test if IDs match. Example error message: "java.lang.AssertionError: JSON path "$[:1].id" expected:<1> but was:<1>"

### CODE REVIEW

File: groupnews-service/src/test/java/com/epam/internship/GroupServiceTest.java [hunk 1]
```diff
@@ -89,4 +101,54 @@ public void shouldFailWhenDescCharacterNumberExceeds2000() {
...
+		when(userRepository.findOne(USER_ID)).thenReturn(userEntity);
+		when(groupRepository.findByCreatedBy(userEntity)).thenReturn(Arrays.asList(groupEntity1, groupEntity2));
+		when(conversionService.convert(groupRepository.findByCreatedBy(userEntity),
+				TypeDescriptor.collection(List.class, TypeDescriptor.valueOf(GroupEntity.class)),
+				TypeDescriptor.collection(List.class, TypeDescriptor.valueOf(Group.class))))
+						.thenReturn(Arrays.asList(group1, group2));
+		// When
+		List<Group> groups = systemUnderTest.listGroupsBelongingToUser(USER_ID);
+		// Then
+		for (int i = 0; i < groups.size(); i++) {
```
Comment (uzonyib) [inline groupnews-service/src/test/java/com/epam/internship/GroupServiceTest.java]: Please refactor is to a single assert. Use an assertion lib if needed.

File: groupnews-web/src/main/java/com/epam/internship/controller/UserController.java [hunk 2]
```diff
@@ -47,4 +47,9 @@ public Group createGroup(@PathVariable("userId") Long userId, @RequestParam Stri
 			@RequestParam String description) {
 		return groupService.createGoup(userId, title, description);
 	}
+	
+	@GetMapping("/{userId}/groups")
+	public List<Group> getAllGroupsBelongingToUser(@PathVariable("userId")Long userId){
```
Comment (uzonyib) [inline groupnews-web/src/main/java/com/epam/internship/controller/UserController.java]: Should go to GroupController.

File: groupnews-web/src/test/java/com/epam/internship/UserControllerTest.java [hunk 3]
```diff
@@ -180,14 +187,50 @@ public void shouldReturnBadRequestWhenGroupTitleIsLongerThanAllowed() throws Exc
...
+	public void shouldReturnListOfGroupsBelongingToUser() throws Exception {
+		// Given
+		Group group1 = Group.builder().id(1L).title(GROUP_TITLE).description(GROUP_DESCRIPION).createdBy(user1).build();
+		Group group2 = Group.builder().id(GROUP_ID_2).title(GROUP_TITLE_2).description(GROUP_DESCRIPION_2)
+				.createdBy(user1).build();
+		when(groupService.listGroupsBelongingToUser(USER_ID_1)).thenReturn(Arrays.asList(group1, group2));
+		// When
+		mockMvc.perform(get("/users/1/groups"))
+				// Then
+				// .andExpect(jsonPath("$[:1].id").value(group1.getId()))
```
Comment (uzonyib) [inline groupnews-web/src/test/java/com/epam/internship/UserControllerTest.java]: Please use group1.getId().intValue().

**Review by uzonyib — APPROVED** [review]

File: groupnews-service/src/test/java/com/epam/internship/GroupServiceTest.java [hunk 4]
```diff
@@ -89,4 +102,52 @@ public void shouldFailWhenDescCharacterNumberExceeds2000() {
...
+				.description(GROUP_DESCRIPTION).createdBy(userEntity).build();
+		GroupEntity groupEntity2 = GroupEntity.builder().id(GROUP_ID_2).title(GROUP_TITLE_2)
+				.description(GROUP_DESCRIPTION_2).createdBy(userEntity).build();
+		User user = User.builder().id(USER_ID).build();
+		Group group1 = Group.builder().id(GROUP_ID_1).title(GROUP_TITLE).description(GROUP_DESCRIPTION).createdBy(user)
+				.build();
+		Group group2 = Group.builder().id(GROUP_ID_2).title(GROUP_TITLE_2).description(GROUP_DESCRIPTION_2)
+				.createdBy(user).build();
+		List<Group> expectedGroups = new ArrayList<>();
+		expectedGroups.addAll(Arrays.asList(group1, group2));
```
Comment (Attila-Neuwirth) [inline groupnews-service/src/test/java/com/epam/internship/GroupServiceTest.java]: You can init expectedGroups with `List<Group> expectedGroups = Arrays.asList(group1, group2);`

**Review by Attila-Neuwirth — APPROVED** [review]

**Pull request merged**